eldritch/test_mythos.py

```diff
@@ -2055,6 +2058,130 @@ def testNormal(self):
     self.assertEqual(self.char.clues, 1)
 
 
+class Mythos18Test(EventTest):
+  def setUp(self):
+    super().setUp()
+    self.mythos = Mythos18()
+    self.state.mythos.append(self.mythos)
+    self.state.event_stack.append(Mythos(self.char))
+    self.resolve_until_done()
+    self.char.sanity = 2
+
+    specials = location_specials.CreateFixedEncounters()
+    for location_name, fixed_encounters in specials.items():
+      self.state.places[location_name].fixed_encounters.extend(fixed_encounters)
+    self.state.places["Downtown"].encounters.extend(
+        encounters.CreateEncounterCards()["Downtown"]
+    )
+
+  def testCanGainFromPsychology(self):
+    self.char.possessions.append(abilities.Psychology())
+    self.advance_turn(0, "upkeep")
+    self.state.event_stack.append(UpkeepActions(self.char))
+    psych = self.resolve_to_usable(0, "Psychology")
+    self.state.event_stack.append(psych)
+    char_choice = self.resolve_to_choice(MultipleChoice)
+    char_choice.resolve(self.state, "Dummy")
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 3)
+
+  def testCanGainAtAsylum(self):
+    self.char.dollars = 2
+    self.char.place = self.state.places["Asylum"]
+    self.advance_turn(0, "encounter")
+    choice = self.resolve_to_choice(CardSpendChoice)
+    choice.resolve(self.state, "Restore 1 Sanity")
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 3)
+    self.advance_turn(1, "encounter")
+    choice = self.resolve_to_choice(CardSpendChoice)
+    self.assertEqual(choice.choices, ["Downtown Card", "Restore 1 Sanity", "Restore All Sanity"])
+    self.spend("dollars", 2, choice)
+    choice = self.resolve_to_choice(CardSpendChoice)
+    choice.resolve(self.state, "Restore All Sanity")
+    self.resolve_until_done()
+    self.assertEqual(self.char.sanity, 5)
+
+  def testCantGainFromOthers(self):
+    self.char.possessions.append(items.HealingStone(0))
```

Comment:
TODO for me: maybe the healing stone shouldn't even let you try to gain sanity if this card is in play.